### ISSUE TITLE
Customizable auth scopes.

### ISSUE THREAD

**Issue by jason-dour** [joyceerhl/vscode-github-graphql-notebooks#9]
Recently, while using the extension to prototype queries, I ran into some hurdles.

Specifically, I was restricted from prototyping several queries related to my organizations.  Did a bit of digging, and figured out that  line 14 of extension.ts is the culprit:

https://github.com/joyceerhl/vscode-github-graphql-notebooks/blob/0216faff8823d2afc33661465bab3c7acffe16af/src/extension.ts#L14

If authentication scopes could be customized, users could authorize scopes other than `repo` and `workflow`.  I made the change in the local cached copy of the extension and confirmed my queries were now functional.

I could attempt to write a fix and send a pull request; just haven't done much work in VSC extensions...

**Comment by joyceerhl** [comment]
Good find, this should definitely be configurable. If you'd like to send a PR for this that would be awesome 😊 I think it would make sense to 1) introduce a setting like `"github-graphql-nb.scopes"` which defaults to `['repo', 'workflow']` but can be configured to whatever scopes you want, and 2) update the extension code to read the scopes out of the workspace configuration when creating a GitHub `AuthenticationSession`.

For the first part, you'd want to modify the package.json file to include a `configuration` entry under the `contributes` section here: https://github.com/joyceerhl/vscode-github-graphql-notebooks/blob/0216faff8823d2afc33661465bab3c7acffe16af/package.json#L28-L46

The entry could look like
```json
"contributes": {
  "configuration": {
    "title": "GitHub GraphQL Notebooks",
    "properties": {
      "github-graphql-nb.scopes": {
        "type": "array",
        "default": ["repo", "workflow"],
        "description": "The authentication scopes to request when signing into GitHub.",
        "source": "resource"
      }
    }
  },
  ...
}
```

For the second part, you can access the value of this setting from the extension code by getting the [`WorkspaceConfiguration`](https://code.visualstudio.com/api/references/vscode-api#WorkspaceConfiguration):

```ts
const scopes = vscode.workspace.getConfiguration('github-graphql-nb').get('scopes');
```


And you'd pass `scopes` into these two calls to `authentication.getSession`: https://github.com/joyceerhl/vscode-github-graphql-notebooks/blob/0216faff8823d2afc33661465bab3c7acffe16af/src/extension.ts#L116 https://github.com/joyceerhl/vscode-github-graphql-notebooks/blob/0216faff8823d2afc33661465bab3c7acffe16af/src/extension.ts#L126

What do you think?

**Comment by jason-dour-gea** [comment]
I'll take a look at it.  Thanks!

**Comment by jason-dour** [comment]
Submitted a PR.  I **think** it covers the need.

**Comment by joyceerhl** [comment]
Thanks so much for working on this @jason-dour 😄 I've published a pre-release with your changes (and also fixed up the npm security issues you highlighted). It's working well for me so far:

<img width="1280" alt="Screen Shot 2022-04-13 at 8 34 40 PM" src="https://user-images.githubusercontent.com/30305945/163308627-94b7f57d-093a-400b-9eac-2b0a9548cea5.png">
 
Would you mind switching to the pre-release of this extension and verifying that it also works well for you? Once you've done that I will go ahead and publish a stable release of this extension with the setting you added. 

You can install the pre-release version from the extension page:
<img width="793" alt="Screen Shot 2022-04-13 at 8 38 04 PM" src="https://user-images.githubusercontent.com/30305945/163309001-c72972d6-1c2f-4d27-bdb6-e1be07910790.png">

**Comment by jason-dour** [comment]
Switched to pre-release version, and was able to confirm a query failed, then added non-default scope, re-authed, and confirmed the query worked.  I'd say good to go.  :)

**Comment by joyceerhl** [comment]
Just published a stable build (version `v0.2.0`) with your contributed setting. I did tweak the name from `github-graphql-nb` to `githubGraphql`—it slipped my mind that since VS Code adds spaces to camelcased configuration names, that ends up looking much nicer in the UI compared to the kebab-cased version that I originally suggested.
<img width="1280" alt="Screen Shot 2022-04-14 at 6 17 25 PM" src="https://user-images.githubusercontent.com/30305945/163502602-82ad6f5a-cd1c-40a6-b9b8-9807ecff8b52.png">

Thanks a ton for spending your time to make this extension better!
